src/components/media-views.js

```diff
@@ -421,6 +427,23 @@ AFRAME.registerComponent("media-video", {
     this.changeVolumeBy(-0.1);
   },
 
+  async snap() {
+    if (this.isSnapping) return;
+    this.isSnapping = true;
+    this.el.sceneEl.systems["hubs-systems"].soundEffectsSystem.playSoundOneShot(SOUND_CAMERA_TOOL_TOOK_SNAPSHOT);
+
+    const canvas = document.createElement("canvas");
+    canvas.width = this.video.videoWidth;
+    canvas.height = this.video.videoHeight;
+    canvas.getContext("2d").drawImage(this.video, 0, 0, canvas.width, canvas.height);
+    const blob = await new Promise(resolve => canvas.toBlob(resolve));
+    const file = new File([blob], "snap.png", { type: "image/png" });
```

Comment:
Could reuse this `{ type: "image/png" }` object by declaring it at the top of the file if you feel so inclined.